File: GenicSolutionAngular/ClientApp/src/app/services/genic-zone.ts
export class GenicZone {
    ZoneId: number;
    ZoneName: string;
    CountryName: string;
    IsActive: boolean;
    IsAssigned: boolean;
    ZoneCoordinatesList: ZoneCoordinates[];
}
export class ZoneCoordinates {
    Id: number;
    ZoneId: number;
    Coordinates: string;
}
export class EngineerWiseZone {
    Id: number;
    ZoneId: number;
    ZoneName: string;
    EngineerId?: string;
}

File: GenicSolution/Entities/GenicZone.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace GenicSolution.Entities
{
    public class GenicZone
    {
        [Key]
        public int ZoneId { get; set; }
        [Column(TypeName = "NVarChar(255)")]
        public string ZoneName { get; set; }
        [Column(TypeName = "NVarChar(255)")]
        public string CountryName { get; set; }
        [DefaultValue(true)]
        public bool IsActive { get; set; }
        [DefaultValue(false)]
        public bool IsAssigned { get; set; }
        public virtual ICollection<ZoneCoordinates> ZoneCoordinatesList { get; set; }
    }
    public class ZoneCoordinates
    {
        [Key]
        public int Id { get; set; }
        [DefaultValue(0)]
        public int ZoneId { get; set; }
        [Column(TypeName = "NVarChar(max)")]
        public string Coordinates { get; set; }
    }
    public class EngineerWiseZone
    {
        [Key]
        public int Id { get; set; }
        [ForeignKey("GenicZone")]
        public int ZoneId { get; set; }
        [Column(TypeName = "NVarChar(255)")]
        public string ZoneName { get; set; }
        [ForeignKey("Engineer")]
        public Guid? EngineerId { get; set; }
    }
}
